src/app/organization/offices/create-office/create-office.component.ts

```diff
@@ -0,0 +1,60 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OrganizationService } from '../../organization.service';
+import { Router, ActivatedRoute } from '@angular/router';
+import { DatePipe } from '@angular/common';
+
+@Component({
+  selector: 'mifosx-create-office',
+  templateUrl: './create-office.component.html',
+  styleUrls: ['./create-office.component.scss']
+})
+export class CreateOfficeComponent implements OnInit {
+
+  officeForm: FormGroup;
+  officeData: any;
+  /** Minimum Due Date allowed. */
+  minDate = new Date(1990, 0, 1);
+  /** Maximum Due Date allowed. */
+  maxDate = new Date();
+
+  constructor(private formBuilder: FormBuilder,
+              private organizationService: OrganizationService,
+              private router: Router,
+              private route: ActivatedRoute,
+              private datePipe: DatePipe) {
+    this.route.data.subscribe((data: { offices: any }) => {
+      this.officeData = data.offices;
+    });
+  }
+
+  ngOnInit() {
+    this.createofficeForm();
+  }
+
+  /*
+  Creates the Office Form
+  */
+  createofficeForm() {
+    this.officeForm = this.formBuilder.group({
+      'name': ['', Validators.required],
+      'parentId': ['', Validators.required],
+      'openingDate': ['', Validators.required],
+      'externalId': [''],
+    });
+  }
+
+  /**
+   * Submits the Office Data.
+   */
+  submit() {
+    const office = this.officeForm.value;
+    office.locale = 'en';
+    office.dateFormat = 'dd MM yyyy';
+    office.openingDate = this.datePipe.transform(office.openingDate, office.dateFormat);
```

Comment:
https://github.com/openMF/web-app/wiki/Date-and-Language

Refer this and use `patchValue`